backend/src/event/event.service.ts

```diff
@@ -0,0 +1,405 @@
+import {
+  BadRequestException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
+import { PrismaService } from '../global/prisma/prisma.service';
+import { EventDTO } from './dto/event.dto';
+import {
+  Equipment,
+  EquipmentsOnEvents,
+  Event,
+  Organizer,
+  User,
+  Venues,
+  VenuesOnEvents,
+} from '@prisma/client';
+import {
+  EventCreateDTO,
+  EventTypeEnum,
+  StatusEnum,
+  ViewAccessEnum,
+} from './dto/event-create.dto';
+import { UserService } from '../user/user.service';
+import { OrganizerService } from '../organizer/organizer.service';
+import { EquipmentService } from '../equipment/equipment.service';
+import { VenueService } from '../venue/venue.service';
+import {
+  NotFoundError,
+  PrismaClientKnownRequestError,
+} from '@prisma/client/runtime';
+import { SortedEquipmentsDTO } from './dto/sorted-equipments.dto';
+import { SortedVenuesDTO } from './dto/sorted-venues.dto';
+import { AvailabilityEnum, ScheduleDTO } from '../schedule/dto/schedule.dto';
+import { ValidationErrorDTO } from '../shared/dto/validation-error.dto';
+import { ErrorFieldDTO } from '../shared/dto/error-field.dto';
+
+@Injectable()
+export class EventService {
+  constructor(
+    private readonly prismaService: PrismaService,
+    private readonly equipmentService: EquipmentService,
+    private readonly venueService: VenueService
+  ) {}
+
+  public async createEvent(dto: EventCreateDTO, user: User): Promise<EventDTO> {
+    if (dto.status === StatusEnum.RESERVED) {
+      throw new BadRequestException(
+        'Cannot have RESERVED status on event creation'
+      );
+    }
+
+    const validationErrorDTO = await this.verifyEventCreation(dto);
+
+    if (validationErrorDTO.hasErrors) {
+      throw new BadRequestException(validationErrorDTO, 'Error creating event');
+    }
+
+    const event = await this.prismaService.event.create({
+      data: {
+        title: dto.title,
+        description: dto.description,
+        status: dto.status,
+        startTime: dto.startTime,
+        endTime: dto.endTime,
+        contactPerson: dto.contactPerson,
+        contactNumber: dto.contactNumber,
+        approvedBy: dto.approvedBy,
+        viewAccess: dto.viewAccess,
+        type: dto.type,
+        additionalNotes: dto.additionalNotes,
+        organizer: {
+          connect: { id: dto.organizerId },
+        },
+        encodedBy: { connect: { id: user.id } },
+        equipments: {
+          create: dto.equipmentIds.map((id) => ({
+            equipmentId: id,
+          })),
+        },
+        venues: {
+          create: dto.venueIds.map((id) => ({
+            venuesId: id,
+          })),
+        },
+      },
+      include: {
+        encodedBy: true,
+        organizer: true,
+        equipments: { include: { equipment: true } },
+        venues: { include: { venue: true } },
+      },
+    });
+
+    await this.addSchedules(dto);
```

Comment:
automatically adds schedules to equipments/venues based on dto's startTime and endTime